### PR TITLE
[CAPI] remove legacy CAPI source code and document

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
 - Remove legacy C API.
 - Update documents.

### Tickets:
 - 128245

### CODE REVIEW

**Review by zhaixuejun1993 — APPROVED** [review]

File: docs/snippets/ov_preprocessing_migration.c [hunk 1]
```diff
@@ -145,11 +145,3 @@ int main_new() {
     return 0;
 }
 
-int main_old() {
```
Comment (vurusovs) [inline docs/snippets/ov_preprocessing_migration.c:148]: Looks like all `migration` files may be removed

@ilya-lavrenov please, comment. If yes, let's remove it in next PR

File: src/bindings/c/README.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ OpenVINO C API has the following structure:
  * [src](./src) contains the implementations of all C APIs.
  * [tests](./tests) contains all tests for OpenVINO C APIs. [Learn more](./docs/how_to_write_unit_test.md).
 
-> **NOTE**:  Using API 2.0 is strongly recommended. Legacy API (for C) [header file](./include/c_api/ie_c_api.h), [source file](./src/ie_c_api.cpp), [unit test](./tests/ie_c_api_test.cpp) are also included in the component, but the legacy API is no longer extended. 
+> **NOTE**:  Using API 2.0 is strongly recommended. Legacy API (for C) has been removed from 2024.0.
```
Comment (vurusovs) [inline src/bindings/c/README.md:32]:
```suggestion
> **NOTE**:  Legacy Inference Engine API (for C) has been removed from 2024.0. 
```

**Review by vurusovs — APPROVED** [review]

File: src/bindings/c/docs/api_overview.md [hunk 3]
```diff
@@ -1,327 +1,410 @@
...
+
+OPENVINO_C_VAR(const char*) ov_property_key_intel_gpu_dev_object_handle;
+
+OPENVINO_C_VAR(const char*) ov_property_key_intel_gpu_va_plane;
+```
+
+
+## OV Core
+
+This strcut represents OpenVINO entity and allows you to manipulate with plugins using unified interfaces.
```
Comment (vurusovs) [inline src/bindings/c/docs/api_overview.md:374]:
```suggestion
This struct represents OpenVINO entity and allows you to manipulate with plugins using unified interfaces.
```

File: src/bindings/c/docs/api_overview.md [hunk 4]
```diff
@@ -342,453 +425,673 @@ This strcut represents an Inference Engine entity and allows you to manipulate w
...
+      printf("build number: %s \n", version.buildNumber);
+      ov_version_free(&version);
     ```
 
-- `IEStatusCode ie_core_load_network(ie_core_t *core, ie_network_t *network, const char *device_name,  ie_config_t config, ie_executable_network_t *exec_network_result)`
+- `ov_status_e ov_core_read_model(const ov_core_t* core, const char* model_path, const char* bin_path, ov_model_t** model)`
 
-  - Description: Loads a network that was read from the Intermediate Representation (IR) to the plugin with specified device name and creates an `ie_executable_network_t` instance of the `ie_network_t` struct.
-    You can create as many networks as you need and use them simultaneously (up to the limitation of the hardware resources).
+  - Description: Reads models from IR / ONNX / PDPD / TF / TFLite formats to create ov_model_t.
```
Comment (vurusovs) [inline src/bindings/c/docs/api_overview.md:452]:
```suggestion
  - Description: Reads models from IR / PyTorch / ONNX / PDPD / TF / TFLite formats to create ov_model_t.
```

Comment (riverlijunjie) [reply src/bindings/c/docs/api_overview.md:452]: Will update these documentation issued in the next PR.

File: docs/snippets/ov_preprocessing_migration.c (hunk 1 above)
Comment (ilya-lavrenov) [reply docs/snippets/ov_preprocessing_migration.c:148]: yes, they should be removed

**Pull request merged**